tests/unit/generate_detect_target_contour.py

```diff
@@ -0,0 +1,192 @@
+        """
+        self.image = image
+
+
+class BoundingBox:
+    """
+    Holds the data that define the generated bounding boxes.
+    """
+    def __init__(self, top_left: tuple[int, int], bottom_right: tuple[int, int]):
+        """
+        top_left: x, y coordinates representing the top left corner of the bounding box on an image.
+        bottom_right: x, y coordinates representing the bottom right corner of the bounding box on an image.
+        """
+        self.top_left = top_left
+        self.bottom_right = bottom_right
+
+
+class InputImageAndExpectedBoundingBoxes:
+    '''
+    Struct to hold the data needed to perform the tests.
+    '''
+    def __init__(self, image: np.ndarray, boxes_list: np.ndarray):
+        """
+        image = A numpy array that represents the image needed to be tested.
+        bounding_box_list: A numpy array that holds a list of expected bounding box coordinates.
+        """
+        self.image = image
+        self.bounding_box_list = boxes_list
+
+
+def add_blurred_landing_pad(
+    background: np.ndarray, landing_data: LandingPadImageConfig
+) -> NumpyImage:
+    """
+    Blurs an image and adds a singular lading pad to the background.
+
+    background: A numpy image.
+    landing_data: Landing pad data for the landing pad to be blurred and added.
+
+
+    Returns: Image with the landing pad.
+    """
+    x, y = background.shape[:2]
+
+    mask = np.zeros((x, y), np.uint8)
+    mask = cv2.ellipse(
+        mask,
+        landing_data.center,
+        landing_data.axis,
+        landing_data.angle,
+        0,
+        360,
+        255,
+        -1,
+    )
+
+    mask = cv2.blur(mask, (25, 25), 7)
+
+    alpha = mask[:, :, np.newaxis] / 255.0
+    # Brings the image back to its original colour.
+    fg = np.full(background.shape, LANDING_PAD_COLOUR_BLUE, dtype=np.uint8)
+
+    blended = (background * (1 - alpha) + fg * alpha).astype(np.uint8)
+    return NumpyImage(blended)
+
+
+def draw_landing_pad(
+    image: np.ndarray, landing_data: LandingPadImageConfig
+) -> tuple[NumpyImage, BoundingBox]:
+    """
+    Draws a single landing pad on the provided image and saves the bounding box coordinates to a text file.
+
+    image: The image to add a landing pad to.
+    landing_data: Landing pad data for the landing pad to be added.
+
+    Returns: Image with landing pad and the bounding box for the drawn landing pad.
+    """
+    (h, k), (a, b) = landing_data.center, landing_data.axis
+    angle_in_rad = math.radians(landing_data.angle)
+    ux, uy = a * math.cos(angle_in_rad), a * math.sin(angle_in_rad)
+    vx, vy = b * math.sin(angle_in_rad), b * math.cos(angle_in_rad)
+    width, height = 2 * math.sqrt(ux**2 + vx**2), 2 * math.sqrt(uy**2 + vy**2)
+
+    top_left = (int(max(h - (0.5) * width, 0)), int(max(k - (0.5) * height, 0)))
+    bottom_right = (
+        int(min(h + (0.5) * width, image.shape[1])),
+        int(min(k + (0.5) * height, image.shape[0])),
+    )
```

Comment:
Why are these being cast to `int` ? Pixel coordinates don't need to be aligned to a grid.